tests/std/tests/P0024R2_parallel_algorithms_for_each/test.cpp

```diff
@@ -47,27 +47,35 @@ const auto call_only_once = [](atomic<bool>& b) { assert(!b.exchange(true)); };
 
 const auto atomic_identity = [](atomic<bool>& b) { return b.load(); };
 
-template <template <class...> class Container>
-void test_case_for_each_parallel(const size_t testSize) {
+template <typename ExecutionPolicy, template <class...> class Container>
+void test_case_for_each_parallel(const size_t testSize, const ExecutionPolicy& exec) {
     Container<atomic<bool>> c(testSize);
-    for_each(par, c.begin(), c.end(), call_only_once);
+    for_each(exec, c.begin(), c.end(), call_only_once);
     assert(all_of(c.begin(), c.end(), atomic_identity));
 }
 
-template <template <class...> class Container>
-void test_case_for_each_n_parallel(const size_t testSize) {
+template <typename ExecutionPolicy, template <class...> class Container>
+void test_case_for_each_n_parallel(const size_t testSize, const ExecutionPolicy& exec) {
     Container<atomic<bool>> c(testSize);
-    auto result = for_each_n(par, c.begin(), testSize, call_only_once);
+    auto result = for_each_n(exec, c.begin(), testSize, call_only_once);
     assert(result == c.end());
     assert(all_of(c.begin(), c.end(), atomic_identity));
 }
 
 int main() {
     test_case_for_each_n();
-    parallel_test_case(test_case_for_each_parallel<forward_list>);
-    parallel_test_case(test_case_for_each_parallel<list>);
-    parallel_test_case(test_case_for_each_parallel<vector>);
-    parallel_test_case(test_case_for_each_n_parallel<forward_list>);
-    parallel_test_case(test_case_for_each_n_parallel<list>);
-    parallel_test_case(test_case_for_each_n_parallel<vector>);
+    parallel_test_case(test_case_for_each_parallel<parallel_policy, forward_list>, par);

```

Comment:
I don't get it: why pass the policy both as a type and as a value, when either would suffice?

Reply:
To pass it only as a type we would need special cases in `parallel_test_case` shared by a fleet of tests and I don't want to do that. It can't be passed only as a value because `parallel_test_case` itself needs to be able to deduce the type of its first parameter and it can't do that without the exact overload being named here.